packages/data-provider/test/utils/convertSqonToQuery.spec.ts

```diff
@@ -0,0 +1,250 @@
+import { expect } from 'chai';
+import { SQL, SQLChunk } from 'drizzle-orm';
+import { describe, it } from 'mocha';
+
+import { SQON } from '@overture-stack/sqon-builder';
+
+import { convertSqonToQuery, parseSQON } from '../../src/utils/convertSqonToQuery.js';
+
+/**
+ * Function to facilitate test cases to extract array of SQL chunks from a `SQL` object
+ * @param {SQL | undefined} obj
+ * @param {string} key
+ * @returns {SQLChunk[]}
+ */
+function extractValues(obj: SQL | undefined, key: string) {
+	let values: SQLChunk[] = [];
+
+	function recurse(currentObj: SQL | undefined) {
+		if (Array.isArray(currentObj)) {
+			currentObj.forEach(recurse);
+		} else if (currentObj && typeof currentObj === 'object') {
+			const objectDescriptor = Object.getOwnPropertyDescriptor(currentObj, key);
+			if (objectDescriptor?.value) {
+				values = values.concat(objectDescriptor.value);
+			}
+			Object.values(currentObj).forEach(recurse);
+		}
+	}
+
+	recurse(obj);
+	return values;
+}
+
+describe('Test SQON functions', () => {
+	describe('Test SQON with greater than filter', () => {
+		const sqonGreaterThanFilterRawInput = {
+			op: 'and',
+			content: [
+				{
+					op: 'gt',
+					content: {
+						fieldName: 'date_of_birth',
+						value: 197005,
+					},
+				},
+			],
+		};
+
+		const sqonGreaterThanFilterParsed: SQON = {
+			op: 'gt',
+			content: { fieldName: 'date_of_birth', value: 197005 },
+		};
+
+		const greaterThanFilterChunk: SQLChunk[] = ["data ->> 'date_of_birth' > '197005'"];
+
+		it('should convert a json text with greater than filter into a SQON format', () => {
+			const result = parseSQON(sqonGreaterThanFilterRawInput);
+			expect(JSON.stringify(result)).to.eql(JSON.stringify(sqonGreaterThanFilterParsed));
+		});
```

Comment:
`parseSQON` appears to just be a wrapper over the SQONBuilder library... it doesn't really need test cases. If there are specific requirements you need this to perform then certainly write those. This test just checks if the outer `and` combination operator is stripped from the SQON...

Reply:
`parseSQON` is a wrapper of a SQONBuilder function and we're not adding any logic. Will remove this test .